app/controllers/api/teams.php

```diff
@@ -787,6 +811,14 @@
             throw new Exception(Exception::TEAM_NOT_FOUND);
         }
 
+        /**
+         * Force document security
+         */
+        $validator = new Authorization('delete');
```

Comment:
It's tough because this is the one case where we want document-only permissions, contrary to the new permission model in general. I can't think of a good way to do it other than handling it per case